wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -1232,6 +1233,8 @@ extension ZMUserSession: SyncAgentDelegate {
 
         WaitingGroupTask(context: syncContext) { [weak self] in
             guard let self else { return }
+            try Task.checkCancellation()
```

Comment:
going in background and foreground, this is executed multiple times, if the enclosing task is cancelled we should return early too